main/java/com/example/myapplication/data/UserRepository.java

```diff
@@ -0,0 +1,32 @@
+package com.example.myapplication.data;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+
+import java.lang.invoke.MutableCallSite;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public class UserRepository {
+    private final ArrayList<User> users = new ArrayList<>();
+    private final MutableLiveData<List<User>>userLiveData = new MutableLiveData<>(users);
+
+    @Inject
+    public UserRepository(){}
+
+    public void addUser(User user){
+        users.add(user);
+        userLiveData.setValue(users);
```

Comment:
C'est pas très grave, mais il fallait probablement que tu envoie new ArrayList<>(users) pour que ça fonctionne.
```suggestion
        userLiveData.setValue(new ArrayList<>(users));
```